src/main/java/com/researchspace/webapp/controller/repositories/DigitalCommonsDataUIConnectionConfig.java

```diff
@@ -0,0 +1,70 @@
+package com.researchspace.webapp.controller.repositories;
+
+import static com.researchspace.service.IntegrationsHandler.DIGITAL_COMMONS_DATA_APP_NAME;
+
+import com.researchspace.model.User;
+import com.researchspace.model.oauth.UserConnection;
+import com.researchspace.properties.IPropertyHolder;
+import com.researchspace.service.UserConnectionManager;
+import com.researchspace.webapp.integrations.digitalcommonsdata.DigitalCommonsDataController;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.security.Principal;
+import java.util.Optional;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.cxf.common.security.SimplePrincipal;
+import org.springframework.validation.support.BindingAwareModelMap;
+
+@Slf4j
+public class DigitalCommonsDataUIConnectionConfig implements RSpaceRepoConnectionConfig {
+
+  private User subject;
+  private UserConnectionManager userConnectionManager;
+  private DigitalCommonsDataController digitalCommonsDataController;
+  private IPropertyHolder propertyHolder;
+  private static final String FAKE_ID = "FAKE_ID";
+
+  public DigitalCommonsDataUIConnectionConfig(
+      DigitalCommonsDataController digitalCommonsDataController,
+      UserConnectionManager source,
+      User subject,
+      IPropertyHolder propertyHolder) {
+    this.subject = subject;
+    this.userConnectionManager = source;
+    this.propertyHolder = propertyHolder;
+    this.digitalCommonsDataController = digitalCommonsDataController;
+  }
+
+  @Override
+  public Optional<URL> getRepositoryURL() {
+    try {
+      return Optional.of(new URL(this.propertyHolder.getDigitalCommonsDataBaseUrl()));
+    } catch (MalformedURLException e) {
+      throw new IllegalArgumentException(
+          "Couldn't create Digital Commons Data repositoryURL: " + e.getMessage());
+    }
+  }
+
+  @Override
+  public String getApiKey() {
+    Optional<UserConnection> optUserConnection =
+        userConnectionManager.findByUserNameProviderName(
+            subject.getUsername(), DIGITAL_COMMONS_DATA_APP_NAME);
+    Principal principal = new SimplePrincipal(subject.getUsername());
```

Comment:
the simpler way to create principal object is with:
`Principal principal = () -> subject.getUsername();`

Reply:
changed